mathewDavidov/assignments/mergeIntervals/lc252/lc252.java

```diff
@@ -0,0 +1,57 @@
+
+/**
+ * Problem Statement:
+ * Given an array of intervals representing ‘N’ appointments, 
+ * find out if a person can attend all the appointments.
+ * 
+ * Example 1:
+ * Input: [[0,30],[5,10],[15,20]]
+ * Output: false
+ * 
+ * Example 2:
+ * Input: [[7,10],[2,4]]
+ * Output: true
+ * 
+ * Edge cases:
+ * null / empty array -> return true
+ * one appointment -> return true
+ * 
+ * sort the appointments array based on their starting time
+ * for each interval until the second to last one:
+ *   if the end time of the current interval is greater than the start of the next interval, return false
+ * return true because there is no time conflicts
+ * 
+ * Runtime:
+ * Time: O(n * log n)
+ * Space: O(1)
+ */
```

Comment:
Great work, clear and to the point. I can follow along at each step of your thinking.